File: ignored-test/python-unittest/CustomArgumentException_RecursiveAsyncExceptionWithoutFlattenException.py
import unittest

class TestCustomArgumentException(unittest.TestCase):
    @unittest.skip("")
    def test_recursive_async_exception_without_flatten_exception(self):
        recursion_count = 3

        def inner_action():
            raise Exception("Life is hard")

        def nested_func(count, action):
            if count <= 0:
                return action()
            return nested_func(count - 1, action)

        try:
            nested_func(recursion_count, inner_action)
        except Exception as ex:
            result = str(ex)
            needle_count = result.count("nested_func")
            self.assertTrue(needle_count >= recursion_count, f"{needle_count} too small")

File: ignored-test/java-junit/CustomArgumentException_RecursiveAsyncExceptionWithoutFlattenException.java
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.Disabled;
import static org.junit.jupiter.api.Assertions.assertTrue;

import java.util.concurrent.*;

public class CustomArgumentException {

    @Test
    @Disabled("")
    public void recursiveAsyncExceptionWithoutFlattenException() {
        int recursionCount = 3;
        Callable<Integer> innerAction = () -> { throw new ApplicationException("Life is hard"); };

        Future<Integer> t1 = Executors.newSingleThreadExecutor().submit(() -> {
            return nestedFunc(recursionCount, innerAction);
        });

        try {
            t1.get();
        } catch (ExecutionException ex) {
            ExceptionLayoutRenderer layoutRenderer = new ExceptionLayoutRenderer();
            layoutRenderer.setFormat("ToString");
            layoutRenderer.setFlattenException(false);

            LogEventInfo logEvent = LogEventInfo.create(LogLevel.ERROR, null, null, ex);
            String result = layoutRenderer.render(logEvent);

            int needleCount = 0;
            int foundIndex = result.indexOf("nestedFunc");
            while (foundIndex >= 0) {
                needleCount++;
                foundIndex = result.indexOf("nestedFunc", foundIndex + "nestedFunc".length());
            }

            assertTrue(needleCount >= recursionCount, needleCount + " too small");
        }
    }
}
